tests/rptest/tests/acls_test.py

```diff
@@ -6,38 +6,66 @@
 # As of the Change Date specified in that file, in accordance with
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0
+import socket
+from ducktape.mark import matrix
 from rptest.tests.redpanda_test import RedpandaTest
 from rptest.services.cluster import cluster
 from rptest.services.admin import Admin
 from rptest.clients.rpk import RpkTool, ClusterAuthorizationError
-from rptest.services.redpanda import SecurityConfig
+from rptest.services.redpanda import SecurityConfig, TLSProvider
+from rptest.services import tls
+
+
+class MTLSProvider(TLSProvider):
+    def __init__(self, tls):
+        self.tls = tls
+
+    @property
+    def ca(self):
+        return self.tls.ca
+
+    def create_broker_cert(self, redpanda, node):
+        assert node in redpanda.nodes
+        return self.tls.create_cert(node.name)
+
+    def create_service_client_cert(self, _, name):
+        return self.tls.create_cert(socket.gethostname(), name=name)
 
 
 class AccessControlListTest(RedpandaTest):
     password = "password"
     algorithm = "SCRAM-SHA-256"
 
     def __init__(self, test_context):
+        self.cert = None
+
         security = SecurityConfig()
         security.enable_sasl = True
-        super(AccessControlListTest,
-              self).__init__(test_context,
-                             num_brokers=3,
-                             security=security,
-                             extra_node_conf={'developer_mode': True})
```

Comment:
Is this enabled elsewhere?

Reply:
can you clarify what you are referring to? you mean developer_mode? if so, I removed it because it looks like it was old cruft that wasn't necessary for the test.

Reply:
Yes, `developer_mode`.

Reply:
yeh seems like something ancient